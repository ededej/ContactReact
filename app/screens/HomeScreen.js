import React, { Component } from 'react';
import { FlatList, Button, StyleSheet, Text, View, ScrollView } from 'react-native';
import SimpleAsyncStorage from '../persitence/SimpleAsyncStorage'
import { List, ListItem } from 'react-native-elements';


class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Contacts',
    headerStyle: {
      backgroundColor: 'skyblue',
    }
  }
  constructor(props) {
    super(props)
    this.state = {
      isMounted: false,
      data: [],
      loading: false,
      error: null,

    }

  }
  componentDidMount() {
    this.setState({ isMounted: true })
    this.getContactList();
    this.props.navigation.addListener('willFocus', (playload) => {
      this.getContactList();
    });
  }
  componentWillUnmount() {
    this.setState({ isMounted: false })
    this.props.navigation.removeEventListener('willFocus', (playload) => {
      this.getContactList();
    });
  }
  getContactList = () => {
    this.setState({ loading: true });
    SimpleAsyncStorage.retrieveAllData()
      .then((response) => {

        let tempArray = [];

        if (response != null) {
          console.log('returned response:', response);
          response.map(contact =>
            tempArray.push({
              key: contact[0],
              value: JSON.parse(contact[1])
            }));
          if (this.state.isMounted) {

            this.setState({
              data: tempArray,
              loading: false,
              error: response.error || null
            });
          }

        } else {
          this.setState({ error, loading: false });
        }
      })
      .catch(error => {
        this.setState({ error, loading: false });
      });
  };
  renderSeparator = () => {
    return (
      <View
        style={{
          height: 3,
          backgroundColor: "#CED0CE",
        }}
      />
    );
  };

  viewContact = (contact) => {
    console.log('Contact select', contact);
    this.props.navigation.navigate('ViewContact', { contact });

  }

  _renderItem = ({ item }) => (
    <ListItem
      title={`${item.value.firstName} ${item.value.lastName}`}
      subtitle={item.value.email}
      containerStyle={{ borderBottomWidth: 0 }}
      onPress={() => this.viewContact(item.value)}
    />
  );

  render() {

    return (
      <View style={{ flex: 1 }}>
        <View style={{
          flex: 1,
          backgroundColor: 'skyblue',
          flexDirection: 'row',
          justifyContent: 'space-between',
        }} >
          <Text style={styles.welcome}>Your Contacts</Text>

          <View style={styles.buttonContainer}>

            <Button
              onPress={() => this.props.navigation.navigate('AddContact')}
              title="Add"
            />
          </View>
        </View>

        <View style={{ flex: 11, backgroundColor: 'powderblue' }} >
          <ScrollView >

            <List
              containerStyle={{ borderTopWidth: 0, borderBottomWidth: 0 }}>
              <FlatList
                data={this.state.data}
                ItemSeparatorComponent={this.renderSeparator}
                renderItem={this._renderItem}
              />
            </List>
          </ScrollView >

        </View>

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  welcome: {
    fontSize: 20,
    padding: 10,
  },

  buttonContainer: {
  },
  formStyle: {
    fontSize: 25
  }
});
export default HomeScreen;
