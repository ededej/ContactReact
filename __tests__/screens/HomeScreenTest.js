import React from 'react';
import HomeScreen from '../../app/screens/HomeScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

