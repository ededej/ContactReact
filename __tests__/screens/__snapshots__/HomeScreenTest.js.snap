// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "skyblue",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "padding": 10,
        }
      }
    >
      Your Contacts
    </Text>
    <View
      style={Object {}}
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "powderblue",
        "flex": 11,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#bbb",
                "borderTopWidth": 1,
                "marginTop": 20,
              },
              Object {
                "borderBottomWidth": 0,
                "borderTopWidth": 0,
              },
            ]
          }
        >
          <RCTScrollView
            ItemSeparatorComponent={[Function]}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
